base_domain_operator/models/base_domain_operator.py

```diff
@@ -30,12 +30,19 @@ def _operator_parent_of(self, leaf, expression):
 
         # this function is more of less a verbatim copy of
         # https://github.com/OCA/OCB/blob/10.0/odoo/osv/expression.py#L738
-        def parent_of_domain(left, ids, left_model, parent=None, prefix=''):
+        def parent_of_domain(left, right, left_model, parent=None, prefix=''):
+            if isinstance(right, unicode):
```

Comment:
why don't you just assign `ids` the ids resulting from `name_get` and leave the rest unchanged?

And testing for a string in python < 3 should be done by checking for [`basestring`](https://docs.python.org/2.7/library/functions.html#basestring)